modules/social_features/social_follow_taxonomy/src/Plugin/views/filter/FollowTaxonomyViewsFilter.php

```diff
@@ -147,8 +146,24 @@ protected function valueForm(&$form, FormStateInterface $form_state): void {
               if (!$term->isPublished() && !$this->currentUser->hasPermission('administer taxonomy')) {
                 continue;
               }
+              // There is a PHPStan error:
+              // ---------------------------------------------------------------
+              // Parameter #2 $times of function str_repeat expects int,
+              // Drupal\Core\Field\FieldItemListInterface given.
+              // ---------------------------------------------------------------
+              // The property 'depth' is adding to the term in
+              // TermStorage::loadTree(), this is why we can't load it with the
+              // usual methods.
+              // Typecasting directly for $term->depth is not solve this error.
+              // So, to fix it we need to determine term depth with another
+              // approach.
+              // @todo Load the term's depth in the right way when it will be
+              //   possible
+              $parents = $this->entityTypeManager->getStorage('taxonomy_term')->loadParents((int) $term->id());
+              // The number of parents determines the current depth.
+              $depth = count($parents);
```

Comment:
@Kingdutch Please give this one a look. ^

Reply:
You pointed to `OverviewTerms.php` in Drupal core as an example of where they use `depth` so we can learn from them to make PHPStan happy.

```
// Depth should be added by Drupal core and is used in various places. 
// However it's also not officially documented, so in case Drupal core has not 
// made it available yet we load the depth ourselves and cache it on the term 
// entity.
if (!isset($term->depth)) {
  $parents = $this->entityTypeManager->getStorage('taxonomy_term')->loadParents((int) $term->id());
  $term->depth = count($parents);
}

$choice = new \stdClass();
$choice->option = [$term->id() => str_repeat('-', $term->depth) . $this->entityRepository->getTranslationFromContext($term)->label()];
```